src/main/java/org/javarosa/core/model/trace/EvaluationTraceReporter.java

```diff
@@ -9,12 +9,28 @@
  *
  * Created by ctsims on 10/19/2016.
  */
-public interface EvaluationTraceReporter {
-    boolean wereTracesReported();
+public abstract class EvaluationTraceReporter {
 
-    void reportTrace(EvaluationTrace trace);
+    public abstract boolean wereTracesReported();
 
-    void reset();
+    public abstract void reportTrace(EvaluationTrace trace);
 
-    Vector<EvaluationTrace> getCollectedTraces();
+    public abstract void reset();
+
+    public abstract Vector<EvaluationTrace> getCollectedTraces();
+
+    public void printAndClearTraces(String description) {
```

Comment:
I think moving this functionality to this interface is a bit of a negative. It couples this class strongly with the stringevaluationtracreserializer and moves an interface to an abstract class (which introduces limitations on the upstream classes). The decision of how to output your traces is also not something that the implemented classes would want to change  based on their implementation.

As a practical reason: Will has mentioned that dumping stuff to sysout isn't helpful for formplayer, so he's likely to need to handle the outputs on formplayer differently. Moving this here will make him have to find a way to decouple the output at a class level rather than at the method call level.

Is there a specific reason for moving this here instead of a util method? One other huge advantage of the util method approach is that it lets us make internal decisions based on the type of trace reporter. The null check for the reporter now has to be repeated in a helper method, for instance.